catroid/src/main/java/org/catrobat/catroid/content/actions/WebRequestAction.java

```diff
@@ -0,0 +1,156 @@
+/*
+ * Catroid: An on-device visual programming system for Android devices
+ * Copyright (C) 2010-2018 The Catrobat Team
+ * (<http://developer.catrobat.org/credits>)
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * An additional term exception under section 7 of the GNU Affero
+ * General Public License, version 3, is available at
+ * http://developer.catrobat.org/license_additional_term
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.actions;
+
+import android.support.annotation.VisibleForTesting;
+import android.util.Log;
+
+import com.badlogic.gdx.scenes.scene2d.Action;
+
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.formulaeditor.Formula;
+import org.catrobat.catroid.formulaeditor.InterpretationException;
+import org.catrobat.catroid.formulaeditor.UserVariable;
+import org.catrobat.catroid.stage.StageActivity;
+import org.catrobat.catroid.web.WebConnection;
+import org.catrobat.catroid.web.WebConnectionFactory;
+
+public class WebRequestAction extends Action implements WebConnection.WebRequestListener {
+
+	private static final float TIMEOUT_DURATION = 60f;
+	private static final Double ERROR_TIMEOUT = 504d;
+	private static final Double ERROR_TOO_MANY_REQUESTS = 429d;
+	private static final Double ERROR_BAD_REQUEST = 400d;
+
+	private Sprite sprite;
+	private Formula changeVariable;
+	private String url;
+	private UserVariable userVariable;
+	private WebConnectionFactory webConnectionFactory;
+
+	private boolean firstStart = true;
+	private float time = 0f;
+	private WebConnection webConnection = null;
+	private boolean isRequestFinished = false;
```

Comment:
please use intdefs instead of boolean flags wherever possible. (eg. this could be called RequestStatus.FINISHED, RequestStatus.WAITING, RequestStatus.NOT_SENT, etc...)